h/api/test/auth_test.py

```diff
@@ -102,64 +94,92 @@ def test_userid_from_bearer_token_calls_decode(jwt):
+@userid_from_jwt_fixtures
+def test_userid_from_jwt_returns_None_if_no_sub(jwt):
     jwt.decode.return_value = {}  # No 'sub' key.
 
-    result = auth.userid_from_bearer_token(mock_request(token='Bearer abc123'))
+    result = auth.userid_from_jwt(u'abc123', mock_request())
 
     assert result is None
 
 
-@user_id_from_bearer_token_fixtures
-def test_userid_from_bearer_token_returns_None_if_decoding_fails(jwt):
+@userid_from_jwt_fixtures
+def test_userid_from_jwt_returns_None_if_decoding_fails(jwt):
     jwt.decode.side_effect = InvalidTokenError
 
-    result = auth.userid_from_bearer_token(mock_request(token='Bearer abc123'))
+    result = auth.userid_from_jwt(u'abc123', mock_request())
 
     assert result is None
 
 
-def test_generate_bearer_token_userid_from_bearer_token_successful():
-    """Test generate_bearer_token() and userid_from_bearer_token() together.
+def test_generate_jwt_userid_from_jwt_successful():
+    """Test generate_jwt() and userid_from_jwt() together.
 
-    Test that userid_from_bearer_token() successfully decodes tokens
-    generated by generate_bearer_token().
+    Test that userid_from_jwt() successfully decodes tokens
+    generated by generate_jwt().
 
     """
-    token = auth.generate_bearer_token(mock_request(), 3600)
-    userid = auth.userid_from_bearer_token(
-        request=mock_request(token='Bearer ' + token))
+    token = auth.generate_jwt(mock_request(), 3600)
+    userid = auth.userid_from_jwt(token, mock_request())
 
     assert userid == 'acct:testuser@hypothes.is'
 
 
-def test_generate_bearer_token_userid_from_bearer_token_bad_token():
-    """Test generate_bearer_token() and userid_from_bearer_token() together.
+def test_generate_jwt_userid_from_jwt_bad_token():
+    """Test generate_jwt() and userid_from_jwt() together.
 
-    Test that userid_from_bearer_token() correctly fails to decode a token
-    generated by generate_bearer_token() using the wrong secret.
+    Test that userid_from_jwt() correctly fails to decode a token
+    generated by generate_jwt() using the wrong secret.
 
     """
     request = mock_request()
     request.registry.settings['h.client_secret'] = 'wrong'
-    token = auth.generate_bearer_token(request, 3600)
+    token = auth.generate_jwt(request, 3600)
 
-    userid = auth.userid_from_bearer_token(
-        request=mock_request(token='Bearer ' + token))
+    userid = auth.userid_from_jwt(token, mock_request())
 
     assert userid is None
 
 
+userid_from_api_token_fixtures = pytest.mark.usefixtures('models')
+
+
+@userid_from_api_token_fixtures
+def test_userid_from_api_token_returns_None_when_token_doesnt_start_with_6879(
+        models):
+    assert auth.userid_from_api_token(u'abc123') is None
+    assert not models.Token.get_by_value.called
+
+
+@userid_from_api_token_fixtures
+def test_userid_from_api_token_calls_get_by_value(models):
+    auth.userid_from_api_token(u'6879-abc123')
+
+    models.Token.get_by_value.assert_called_once_with(u'6879-abc123')
```

Comment:
Where at all possible, tests should check only the results of a function not how it is implemented. In this case, I think it suffices to test that the result of `userid_from_api_token` matches whatever the database mock's lookup function returns.